src/lib/sdc/usb-key.sh

```diff
@@ -133,6 +135,45 @@ function unmount_usb_key()
 	umount "$mnt"
 }
 
+# Evil twins of the {,un}mount_usb_key functions.  These are specific
+# to ISO/hsfs (aka. {C,DV,B}D-ROM) disks.  We may be able to factor-out
+# even more common bits from the usb_key functions, but not today.
+
+function mount_ISO
+{
+	local mnt=`extract_mountpath $1`
+
+	# XXX KEBE SAYS JUST TRUST IT FOR NOW...
+	for disk in `disklist -r`; do
```

Comment:
Consider using `diskinfo -p`

Reply:
Doesn't seem to detect CD-ROM/DVD-ROM drives.
```
[root@smartos-bios ~]# mount -F hsfs /dev/dsk/c2t1d0s0 /mnt
[root@smartos-bios ~]# df -h /mnt/.
Filesystem             Size   Used  Available Capacity  Mounted on
/dev/dsk/c2t1d0s0    554.11M 554.11M          0   100%    /mnt
[root@smartos-bios ~]# diskinfo -p
TYPE    DISK                    VID      PID              SIZE          RMV SSD
SCSI    c1t0d0                  VMware,  VMware Virtual S 34359738368   no  yes
SCSI    c1t1d0                  VMware,  VMware Virtual S 34359738368   no  yes
SATA    c2t0d0                  VMware   Virtual SATA Hard Drive 34359738368   no  yes
SATA    c2t2d0                  VMware   Virtual SATA Hard Drive 34359738368   no  yes
[root@smartos-bios ~]# 
```

Reply:
That's unfortunate.. oh well..

Reply:
I do need to ax the KEBE comment here, however.

Reply:
Here, on line 144 prefer `local mnt=$(extract_mountpath "$1")` over backticks.

On line 146, same `mapfile` thing as below in `smartos_config_prompt.sh`

    mapfile -t disks < <(disklist -r)
    for a in "${disks[@]}"; do
        ...
    done